src/AgGrid/ValueFormat.elm

```diff
@@ -167,6 +168,25 @@ numberFilterValueGetter field =
     """ [ field ]
 
 
+{-| Format a cell value as Date.
+
+This is similar to using a normal Value Getter, but is specific to the filter.
+
+    cellValue = "2023-12-12"
+
+    dateFilterValueGetter "dateField"
+    > new Date("2023-12-12T00:00")
+
+-}
+dateFilterValueGetter : String -> String
+dateFilterValueGetter field =
+    String.Interpolate.interpolate """
+        if (!data.{0}) { return null; }
+
+        return new Date(data.{0} + "T00:00")
```

Comment:
Now my only fear is that we might create incorrect DateTime strings if the user provides more than just the date part.

I would say that it is not uncommon for `Date` values to be represented with a POSIX on the frontend and when converting the POSIX to an ISO8601 string for the renderer we would include the time part. This potential for error is not really obvious to the user.

Maybe we can use `setHour` instead (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setHours). This allows us to pass whatever the users provides to the `new Date` call, create a Date from it, and forcibly shift the time to midnight.